app/models/sushi/platform_report.rb

```diff
@@ -75,28 +78,40 @@ def attribute_performance_for_resource_types
       end
     end
 
-    def performance(records)
-      metric_types.each_with_object({}) do |metric_type, hash|
-        hash[metric_type] = records.each_with_object({}) do |record, inner_hash|
-          inner_hash[record.year_month.strftime("%Y-%m")] = record[metric_type.downcase.to_s]
-          inner_hash
-        end
-      end
-    end
-
     def attribute_performance_for_platform
       [{
         "Data_Type" => "Platform",
         "Access_Method" => "Regular",
         "Performance" => {
-          "Searches_Platform" => data_for_platform.each_with_object({}) do |record, hash|
-            hash[record.year_month.strftime("%Y-%m")] = record.total_item_investigations
-            hash
-          end
+          "Searches_Platform" => if granularity == 'Totals'
+                                   total_for_platform = data_for_platform.map(&:total_item_investigations).sum
```

Comment:
Does `data_for_platform.sum(&:total_item_investigations)` work?  If it does that reduces one loop through the data set.

Reply:
let me check!